staging/src/k8s.io/pod-security-admission/policy/check_seccomp_baseline.go

```diff
@@ -0,0 +1,147 @@
+}
+
+func fieldValue(f *field.Path, val string) string {
+	return fmt.Sprintf("%s=%s", f.String(), val)
+}
+
+func fieldValueRequired(f *field.Path) string {
+	return fmt.Sprintf("%s=%s", f.String(), missingRequiredValue)
+}
+
+func CheckSeccompBaseline() Check {
+	return Check{
+		ID:    "seccomp_baseline",
+		Level: api.LevelBaseline,
+		Versions: []VersionedCheck{
+			{
+				MinimumVersion: api.MajorMinorVersion(1, 0),
+				CheckPod:       seccomp_1_0_baseline,
+			},
+			{
+				MinimumVersion: api.MajorMinorVersion(1, 19),
+				CheckPod:       seccomp_1_19_baseline,
+			},
+		},
+	}
+}
+
+func validSeccomp(t corev1.SeccompProfileType) bool {
+	return t == corev1.SeccompProfileTypeLocalhost ||
+		t == corev1.SeccompProfileTypeRuntimeDefault
+}
+
+func validSeccompAnnotation(annotation string) bool {
+	return annotation == corev1.SeccompProfileRuntimeDefault ||
+		strings.HasPrefix(annotation, corev1.SeccompLocalhostProfileNamePrefix)
+}
+
+// baseline policy checks
+
+// seccomp_1_0_baseline checks baseline policy on seccomp alpha annotation
+func seccomp_1_0_baseline(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec) CheckResult {
+	forbidden := sets.NewString()
+
+	if val, ok := podMetadata.Annotations[annotationKeyPod]; ok {
+		if val == corev1.SeccompProfileNameUnconfined {
+			podAnnotationField := field.NewPath("metadata").Child("annotations", annotationKeyPod)
+			forbidden.Insert(fieldValue(podAnnotationField, val))
+		}
+	}
+
+	visitContainersWithPath(podSpec, field.NewPath("spec"), func(c *corev1.Container, path *field.Path) {
+		annotation := annotationKeyContainerPrefix + c.Name
+		if val, ok := podMetadata.Annotations[annotation]; ok {
+			if val == corev1.SeccompProfileNameUnconfined {
+				containerAnnotationField := field.NewPath("metadata").
+					Child("annotations", annotation)
+				forbidden.Insert(fieldValue(containerAnnotationField, val))
+			}
+		}
+	})
+
+	if len(forbidden) > 0 {
+		return CheckResult{
+			Allowed:         false,
+			ForbiddenReason: "seccomp profile",
+			ForbiddenDetail: strings.Join(forbidden.List(), ", "),
+		}
+	}
+
+	return CheckResult{Allowed: true}
+}
+
+// seccomp_1_19_baseline checks baseline policy on securityContext.seccompProfile field
+func seccomp_1_19_baseline(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec) CheckResult {
+	forbidden := sets.NewString()
+
+	if podSpec.SecurityContext != nil {
+		if podSpec.SecurityContext.SeccompProfile != nil {
+			seccompType := podSpec.SecurityContext.SeccompProfile.Type
+			if !validSeccomp(seccompType) {
+				podSeccompField := field.NewPath("spec").Child("securityContext", "seccompProfile", "type")
+				forbidden.Insert(fieldValue(podSeccompField, string(seccompType)))
+			}
+		}
+	}
+
+	visitContainersWithPath(podSpec, field.NewPath("spec"), func(c *corev1.Container, path *field.Path) {
+		if c.SecurityContext != nil {
+			if c.SecurityContext.SeccompProfile != nil {
+				if c.SecurityContext.SeccompProfile.Type != "" {
```

Comment:
can drop this check, if SeccompProfile != nil, type must be valid

Reply:
_Pretty sure_ this caused issues because some fixtures were being generated like

```
securityContext:
  seccompProfile: {}
```

but will triple check.

Reply:
that won't pass API validation:

https://github.com/kubernetes/kubernetes/blob/59e5b849c9439375575f6ced54fb9e2364b58797/pkg/apis/core/validation/validation.go#L3771-L3779

https://github.com/kubernetes/kubernetes/blob/59e5b849c9439375575f6ced54fb9e2364b58797/pkg/apis/core/validation/validation.go#L3823-L3833

if it doesn't pass API validation, it never makes it to PodSecurity admission